### ISSUE TITLE
Allow Vector2 and Vector3 to be serializable by JSON (2960)

### ISSUE THREAD

**Issue by GalacticEmperor1** [pygame-community/pygame-ce#1500]
#  **Issue &numero;2960 opened by *[novialriptide](https://github.com/novialriptide)* at 2021-12-28 05:42:16**

**Description**
If we were to serialize Vector2 or Vector3 right now, it would give out the error: `TypeError: Object of type Vector2 is not JSON serializable`. I think pygame should return the Vectors as tuples when it's being serialized.<hr>

#  Comments

# # #  **[Starbuck5](https://github.com/Starbuck5)* commented at 2021-12-28 05:45:41*

https://stackoverflow.com/questions/3768895/how-to-make-a-class-json-serializable

You're on it, I assume? 👍 
<hr>

# # #  **[novialriptide](https://github.com/novialriptide)* commented at 2021-12-28 06:12:37*

To summarize our Discord conversation about this in # contributing 
It seems like it's impossible to do it at this moment, unless someone opens a PEP to Python about adding a `__json__` method.

An alternative solution to this is to use pickle.
